asv_bench/benchmarks/indexing.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from . import randn, randint
+
+
+nx = 3000
+ny = 2000
+nt = 1000
+ds = xr.Dataset({'var1': (('x', 'y'), randn((nx, ny), frac_nan=0.1)),
+                 'var2': (('x', 't'), randn((nx, nt))),
+                 'var3': (('t', ), randn(nt))},
+                coords={'x': np.arange(nx),
+                        'y': np.linspace(0, 1, ny),
+                        't': pd.date_range('1970-01-01', periods=nt, freq='D'),
+                        'x_coords': ('x', np.linspace(1.1, 2.1, nx))})
+
+basic_indexes = [
+    {'x': slice(0, 3)},
+    {'x': 0, 'y': slice(None, None, 3)},
+    {'x': slice(3, -3, 3), 'y': 1, 't': slice(None, -3, 3)}
+]
+
+basic_assignment_values = [
+    xr.DataArray(randn((3, ny), frac_nan=0.1), dims=['x', 'y']),
+    xr.DataArray(randn(int(ny / 3) + 1, frac_nan=0.1), dims=['y']),
+    xr.DataArray(randn(int((nx - 6) / 3), frac_nan=0.1), dims=['x']),
+]
+
+
+def time_indexing_basic():
+    for ind in basic_indexes:
```

Comment:
Rather than a loop, another option would be to separate these cases into separate benchmarks. That would give more granular information.